### PR TITLE
[Components with more than 1 service] LPS-183545 Refactor ProductNavigationControlMenuEntry components from analytics, layout and segments modules

### PR DESCRIPTION
**Pull request by RenanUmV**
[LPS-183543](https://issues.liferay.com/browse/LPS-183545)

### CODE REVIEW

**Comment by RenanUmV** [comment]
ci:test:analytics-cloud-acceptance

**Comment by zhangyan0701** [comment]
ci:test:echo

**Comment by catarineaguiar** [comment]
no regressions found

File: modules/apps/product-navigation/product-navigation-control-menu-api/src/main/java/com/liferay/product/navigation/control/menu/BaseProductNavigationControlMenuEntry.java [hunk 1]
```diff
@@ -110,6 +111,20 @@ public boolean includeIcon(
 		return false;
 	}
 
+	@Override
+	public boolean isPanelStateOpen(
+		HttpServletRequest httpServletRequest, String key) {
+
+		String layoutReportsPanelState = SessionClicks.get(
+			httpServletRequest, key, "closed");
```
Comment (rafaprax) [inline modules/apps/product-navigation/product-navigation-control-menu-api/src/main/java/com/liferay/product/navigation/control/menu/BaseProductNavigationControlMenuEntry.java:119]: @RenanUmV, rename `layoutReportsPanelState` to `panelState`, please.

File: modules/apps/product-navigation/product-navigation-control-menu-api/src/main/java/com/liferay/product/navigation/control/menu/BaseProductNavigationControlMenuEntry.java [hunk 2]
```diff
@@ -122,4 +137,11 @@ public boolean isUseDialog() {
 		return false;
 	}
 
+	@Override
+	public void setPanelState(
+		HttpServletRequest httpServletRequest, String panelState, String key) {
```
Comment (rafaprax) [inline modules/apps/product-navigation/product-navigation-control-menu-api/src/main/java/com/liferay/product/navigation/control/menu/BaseProductNavigationControlMenuEntry.java:142]: @RenanUmV, please sort the parameters alphabetically. Thx

File: modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/constants/SegmentsExperimentProductNavigationConstants.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ */
+
+package com.liferay.segments.experiment.web.internal.constants;
+
+/**
+ * @author Renan Vasconcelos
+ */
+public class SegmentsExperimentProductNavigationConstants {
+
```
Comment (rafaprax) [inline modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/constants/SegmentsExperimentProductNavigationConstants.java:21]: @RenanUmV, rename to `ProductNavigationControlMenuEntryConstants`, please.

PS: apply to the other constant classes as well. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
liferay/ci:test:relevant

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by RenanUmV** [comment]
ci:test:analytics

**Comment by RenanUmV** [comment]
ci:test:echo

**Comment by RenanUmV** [comment]
ci:test:analytics-cloud-acceptance

**Comment by Kvale1733** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
ci:test:analytics-clout-acceptance

**Comment by Kvale1733** [comment]
ci:test:echo

**Comment by Kvale1733** [comment]
ci:stop

**Comment by Kvale1733** [comment]
@RenanUmV Regression found. Content Performance, Page Audit, and  A/B Test panels do not close after logout

Steps to reproduce:
1. Open any of the mentioned panels
2. Logout and login

Expected Results: The panel is closed
Actual Results: The panel remains open


https://github.com/liferay-core-infra/liferay-portal/assets/90647688/8a6b8753-403e-45f0-8c44-fd6fad26c9f6

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by RenanUmV** [comment]
ci:test:analytics-cloud-acceptance

**Comment by RenanUmV** [comment]
ci:test:echo

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
ci:test:echo

**Comment by zhangyan0701** [comment]
ci:test:analytics-cloud-acceptance

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/product/navigation/control/menu/SegmentsExperimentProductNavigationControlMenuEntry.java [hunk 4]
```diff
@@ -172,9 +177,14 @@ public boolean includeIcon(
...
 	}
 
-	public boolean isPanelStateOpen(HttpServletRequest httpServletRequest) {
+	@Override
+	public boolean isPanelStateOpen(
+		HttpServletRequest httpServletRequest, String key) {
+
 		String segmentsExperimentPanelState = SessionClicks.get(
-			httpServletRequest, _SESSION_CLICKS_KEY, "closed");
+			httpServletRequest,
```
Comment (dantewang) [inline modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/product/navigation/control/menu/SegmentsExperimentProductNavigationControlMenuEntry.java:182]: @RenanUmV Please delegate this part https://github.com/liferay-core-infra/liferay-portal/pull/4599/commits/c814acde4168ac59aba431bd8535392aaf808d17#diff-bfe4e2b1785ce7e685688a7ceb5f83cdea5d6e05714d19d1aa3b923378dddb8cR184-R191 using `super.isPanelStateOpen`:

```
		if (super.isPanelStateOpen(httpServletRequest, key)) {
			return true;
		}
```

**Review by dantewang — CHANGES_REQUESTED** [review]
@RenanUmV Please update, other parts look good to me.

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by RenanUmV** [comment]
ci:test:echo

**Comment by catarineaguiar** [comment]
no regressions found.

**Review by dantewang — APPROVED** [review]

**Comment by dantewang** [comment]
ci:forward:force

cc @shuyangzhou @tinatian

**Pull request closed without merging**